File: java/1_Two_Sum.java
// Solution 01

class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int[] indices = {0, 0};
            
        for(int i = 0; i < nums.length; i++) {
            for (int j = i + 1; j < nums.length; j++) {
                if(nums[i] + nums[j] == target) {
                    indices[0] = i;
                    indices[1] = j;
                }
            }
        }
        
        return indices;
    }
}

// Solution 02

class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        
        for (int i = 0; i < nums.length; i++) {
            
            if (hashMap.containsKey(nums[i])) {
                return new int[]{hashMap.get(nums[i]), i};
            }
            
            hashMap.put(target - nums[i], i);
        }
        
        return new int[0];
    }
}

File: python/1_Two_Sum.py
# Solution 01

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        indices = []
        
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if (nums[i] + nums[j] == target):
                    indices.append(i)
                    indices.append(j)
                    
                    return indices

# Solution 02

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        hashMap = {}
    
        for i, n in enumerate(nums):
            
            difference = target - n
            
            if difference in hashMap:
                return [hashMap[difference], i]
            
            hashMap[n] = i
            
        return
        